examples/tcp-wait/client.rs

```diff
@@ -0,0 +1,265 @@
+        let qr: demi_qresult_t = self.libos.wait(qt, None)?;
+        match qr.qr_opcode {
+            demi_opcode_t::DEMI_OPC_CONNECT => {
+                println!("{} clients connected", num_clients + 1);
+            },
+            demi_opcode_t::DEMI_OPC_FAILED => {
+                // Close socket on error.
+                // FIXME: https://github.com/demikernel/demikernel/issues/640
+                anyhow::bail!("operation failed (qr_ret={:?})", qr.qr_ret)
+            },
+            qr_opcode => {
+                // Close socket on error.
+                // FIXME: https://github.com/demikernel/demikernel/issues/640
+                anyhow::bail!("unexpected result (qr_opcode={:?})", qr_opcode)
+            },
+        }
+        Ok(sockqd)
+    }
+
+    // Attempt to wait for a push() operation to complete after asynchronous closing a socket.
+    pub fn push_async_close_wait(&mut self) -> Result<()> {
+        for i in 0..self.nclients {
+            let sockqd: QDesc = self.connect_to_server(i)?;
+
+            // Close socket on error.
+            // FIXME: https://github.com/demikernel/demikernel/issues/640
+            let push_qt: QToken = self.issue_push(i, sockqd)?;
+            let async_close_qt: QToken = self.libos.async_close(sockqd)?;
+
+            // Wait for async_close().
+            match self.libos.wait(async_close_qt, None) {
+                Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_CLOSE && qr.qr_ret == 0 => {},
+                Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_FAILED && qr.qr_ret == libc::ECONNRESET => {},
+                Ok(_) => anyhow::bail!("wait() should succeed with async_close()"),
+                Err(_) => anyhow::bail!("wait() should succeed with async_close()"),
+            }
+
+            // Wait for push().
+            match self.libos.wait(push_qt, None) {
+                Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_PUSH && qr.qr_ret == 0 => {},
+                _ => anyhow::bail!("wait() should succeed with push() after async_close()"),
+            }
+        }
+
+        Ok(())
+    }
+
+    // Attempt to wait for a pop() operation to complete after asynchronous closing a socket.
+    pub fn pop_async_close_wait(&mut self) -> Result<()> {
+        for i in 0..self.nclients {
+            let sockqd: QDesc = self.connect_to_server(i)?;
+
+            // If error, close sockets.
+            // FIXME: https://github.com/demikernel/demikernel/issues/640
+            let pop_qt: QToken = self.libos.pop(sockqd, None)?;
+            let async_close_qt: QToken = self.libos.async_close(sockqd)?;
+
+            // Wait for async_close().
+            match self.libos.wait(async_close_qt, None) {
+                Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_CLOSE && qr.qr_ret == 0 => {},
+                Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_FAILED && qr.qr_ret == libc::ECONNRESET => {},
```

Comment:
When closing, it is not possible that we get `ECONNRESET`.

Reply:
Done.